plugins/mongodb/README.md

```diff
@@ -0,0 +1,269 @@
+
+**Important note:** when connecting to a MongoDB Cluster and not a local instance the retryWrites=true and w=majority options are already set according to MongoDB examples. So for additional options do not include a question mark at the beginning.
+
+```
+connectionOptions('&ssl=true&connectTimeoutMS=1000') // For connecting to MongoDB Clusters
+connectionOptions('?ssl=true&connectTimeoutMS=1000') // For connecting to a local mongod instance
+```
+
+#### URI options
+
+For specifying timeout settings and other options supported by MongoDB add them to the end of the connection string like this:
+```
+&connectTimeoutMS=30000
+```
+Multiple options are seperated by '&', and more information about additional operations can be found in the [Manual](https://www.mongodb.com/docs/v5.2/reference/connection-string/#connection-string-options).
+
+#### Connecting to the cluster
+
+```
+getConnection() := EMBED(mongodb : user(user), password(pwd), server(server), database(dbname),  collection(collname), batchSize(100))
+ENDEMBED; 
+```
+
+For connecting to a local MongoDB instance you just need to pass in the port number that the server is listening on.
+
+```
+getConnection() := EMBED(mongodb : port(port), database(dbname),  collection(collname))
+ENDEMBED; 
+```
+
+### Parameters
+
+To use function parameters within the MongoDB statement, prefix them with a '$' inside the embedded script.
+
+```
+dataset({STRING _id}) getCount(REAL salary) := EMBED(mongodb : user(user), password(pwd), server(server), database(dbname),  collection(collname), batchSize(100))
+    find({ salary: { $gt: $salary}});
+ENDEMBED; 
+```
+
+The '$' is also reserved in MongoDB, but it does not interfere with the parameters as long as you don't also use a word reserved for a MongoDB command.
+
+### Limitations
+
+There are a few limitations placed by MongoDB on the amount of data you are allowed to transer if you are using a shared cluster. In a single 7-day rolling period you are allowed to transfer:
+
+* M0: 10 GB in and 10 GB out per period
+* M2: 20 GB in and 20 GB out per period
+* M5: 50 GB in and 50 GB out per period
+
+For a more detailed list of the limitations on MongoDB clusters: [Manual](https://www.mongodb.com/docs/atlas/reference/free-shared-limitations/)
+
+### Type conversions
+
+Not every ECL or MongoDB datatype translates seemlessly to the other side.
+
+| ECL datatypes | MongoDB equivalent |
+| ------------- | ------------------ |
+| BOOLEAN | b_bool |
+| REAL | b_double |
+| INTEGER | b_int64 |
+| UINTEGER | b_int64 |
+| UTF8 | b_utf8 |
+| QSting, VString, String | b_utf8 |
+| DECIMAL | b_decimal128|
+
+| MongoDB datatypes | ECL equivalent |
+| ----------------- | -------------- |
+| b_date | STRING, INTEGER |
+| b_regex |  Unsupported |
+| b_timestamp | Unsupported |
```

Comment:
It can be easily mapped to an integer, but not without losing some of the information. This is the format of the timestamp EJSON type: {"$timestamp":{"t":1565545664,"i":1}}. As you can see there are two fields that make up the value for timestamp and there isn't any clear documentation to what that "i" field is for. I didn't want to truncate it without understanding what the purpose of it is.

Reply:
I read somewhere it is a combination of a 32bit number of seconds since 1970 and a 32 bit sequence number.  I think it is fine to not support it for the moment.